packages/wonder-blocks-form/src/components/text-field.tsx

```diff
@@ -44,7 +48,7 @@ type CommonProps = AriaProps & {
      * This `disabled` prop will also set the `readonly` attribute to prevent
      * typing in the field.
      */
-    disabled: boolean;
+    disabled?: boolean;
```

Comment:
I made `disabled`, `light`, and `type` props optional to address typescript errors: https://github.com/Khan/wonder-blocks/actions/runs/11713064887/job/32625094179

These [3 props used to have default props defined](https://github.com/Khan/wonder-blocks/pull/2356/files#diff-3f6913d896b8a5fb2855cf22cd787696ab51bb9833024d46344da4b4de3dbeccL179-L183) in the class component. I'm curious, how did the types work before if they weren't marked as optional? 🤔

Reply:
I believe the TypeScript types for class-based components combine the type of the props passed to `React.Component<>` with the inferred types for `static defaultProps` to determine what the final props should be for the component.

Reply:
I created a TS playground illustrating the difference in types when there is (or isn't) a static `defaultProps` field: https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQdMAnmFnAArFjoC8cA3nTjC4BCBABccVDCjAAdgHMANEJEAjZFCnyAriHVYoqgL706uADZp0AYRJgAjHCwAPGFnkATdNjwwAOntwCHlPGAAeTghuAD4BNWEZZBhgXDgvLAJkXUsYaO44PkERUtFxKTIACyxLSwgyVVKTRLhKbyMACgBKBLKRShhdKHk4PTr6ZroWi2tUOwcAJhd3Tx9MKkDgyDD5SILUeJKBta7e4-7B4dHxy0mRFpncUJkUIrgI7edNKB5+AFYTHAAPSxejPeSvdTvT4OZxiCB-MgAd2gli8ZCBPz+gJBYIYENerhh22W2IBQNB4Je8CYJKW5UR-BRaIxWK0OMp+KAA.